### DISCUSSION TITLE
MET User Config File Information

### DISCUSSION THREAD

**Discussion by Jimjimjiminijim** [dtcenter/METplus#1384] [Configuration] [answered]
Hi there,

I am incredibly new to MET and meteorological modelling (about 4 days in). I am working on a project with a very short time span and am wondering if anyone could help point me in the direction of some more detailed information on configuration files. I really need a list of all possible values that can be accepted by all fields and what they mean... but specifically "model" and "fcst = [{level = }]" would help a great deal. 

I am currently trying to test two identicle .nc files containing total daily precipitation data covering one month with point_stat as a test case to get the software working. I have a bunch of other variables to consider for the overall project such as temperature and climatology data.  

I have read the documentation (https://met.readthedocs.io/en/latest/Users_Guide/config_options.html) entirely but am unable to find the info that I need. It assumes a certain amount of knowledge and I am starting from scratch. I have also had a look through the github but couldn't see anything that answered my questions.

Any help would be greatly appreciated!

Thank you for reading.

Jamie

**Comment by j-opatz** [comment, marked as answer]
Hi Jamie,

And thank you for your question. I understand the overwhelmed feeling of starting with MET all too well: it can be very daunting and can require a lot of reading, walking away from the software, coming back and hoping it makes more sense with some time and distance...it's most certainly a process. But with some time and trial and error learning, I have no doubt that you'll be able to get a decent grasp of the tools and find how to accomplish your verification needs.

You've started things off right by reading through the documentation, specifically the configuration file overview section. Let's start with the `model` variable and `fcst` dictionary, as you indicated.

`Model` is one of the straight forward settings, as it is strictly a variable the user sets to denote what model is being evaluated. It is entirely optional, and only appears in the ASCII output files. Taking a look at the user's guide entry for `model`, we see an example of 

`model = "WRF";`

So this variable is a simple string setting. Note that in the description of this variable, we also see the restrictions for `model`:

`...(no whitespace)...`

And that's it! You've learned your first simple setting in MET. 

The `fcst` dictionary is a bit more complicated, but one of the essential parts of a MET configuration file. I'll save the in-depth discussion of this dictionary for the user's guide, but let me highlight some of the important aspects.

In the user's guide, we're provided an example of what a completed `fcst` dictionary could look like:

`fcst = {
   censor_thresh = [];
   censor_val    = [];
   cnt_thresh    = [ NA ];
   cnt_logic     = UNION;
   wind_thresh   = [ NA ];
   wind_logic    = UNION;
   eclv_points   = 0.05;
   message_type  = [ "ADPSFC" ];
   init_time     = "20120619_12";
   valid_time    = "20120620_00";
   lead_time     = "12";

   field = [
      {
        name       = "APCP";
        level      = [ "A03" ];
        cat_thresh = [ >0.0, >=5.0 ];
      }
   ];
}`

That may seem like a whole mess of configurable things, but there's two important things to remember when looking at this example: first, this contains a large amount of the configuration options available to `fcst` to demonstrate what *can* be set, not necessarily what *needs* to be set. Second, when you focus in on individual entries (`name`, `wind_logic`, etc.) you'll see that while the data type expected might be different, these all boil down to a similar setup as `model`: just variables being set. Let's quickly look at the essential part of the `fcst` dictionary: the `field` entry.

As described, `field` is an array of dictionaries, with each dictionary providing an opportunity for a different forecast field you're interested in verifying. The user's guide lists off some of the more important ones:

`The “name” entry specifies a name for the field.

The “level” entry specifies level information for the field.

Setting “name” and “level” is file-format specific. See below.

The “prob” entry in the forecast dictionary defines probability information. It may either be set as a boolean (i.e. TRUE or FALSE) or as a dictionary defining probabilistic field information.`

So we see that `name` is how MET will access a given file to look for a variable by name, `level` will be the level of the variable name MET pulls (which will differ depending on what type of file you're passing to MET), and `prob` is only necessary if you're dealing with probability fields. For the previous example, we'll also look at the `cat_thresh` entry:

`The “cat_thresh” entry is an array of thresholds to be used when computing categorical statistics.`

Now we have all the necessary information to decode the field entry of the example! It's looking at the APCP (accumulated precipitation) variable at the A03 (accumulated over 3 hours) level using two thresholds for the forecast field: greater than 0.0, and greater than 5.0. 

By going through the rest of the configuration file entries, you'll be able to similarly understand all of the other settings in that bigger example. One thing that may help you as you continue through MET is by stepping through the tutorial. You can find a link[ here](https://dtcenter.org/metplus-practical-session-guide-version-4-0/session-1-metplus-setupgrid-grid). That will walk you through some of the common tools and their capabilities, some of the settings in those tools, as well as an introduction to METplus, which is a Python wrapper around MET that allows users to chain multiple tools together, among other capabilities.

It might also be beneficial to review the online tutorial videos that we've been presenting over the past few weeks. Here's a [link](https://dtcenter.org/events/2021/metplus-training-series/agenda) to the agenda, which has videos embedded for each session.

I really hope these tools help you on your way to learning MET and METplus. Hopefully this answered your questions and if not, please let me know and I can try to find some more examples for you.

**Reply by Jimjimjiminijim** [reply]
Thank you very much for your detailed reply and links to other resources. You have helped me a great deal.
